### PR TITLE
FFI for property names

### LINKED ISSUE #3063: Supporting runtime property use cases
Regex engines like those in the Dart Engine (and Spidermonkey) need to be able to work with properties that have been selected at runtime, from a property string.

This requires the ability to get a unicode set given a property name (and optional value), which may be strings.

ICU4X users can do this manually but they'll have to deal with added properties manually, which isn't great since it doesn't let you update to newer property values in a data driven way.

This will probably need:

 - A mapping for property aliases
 - A mapping from property value names (and aliases) to their indices
 - Potentially some kind of AnyProperty type that works like AnyCalendar

cc @iainireland @echeran @sffc

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes #3063

### CODE REVIEW

**Review by robertbastian — APPROVED** [review]

**Comment by Manishearth** [comment]
Will merge since I have other FFI stuff in the works, but happy to address followups

File: ffi/diplomat/src/properties_names.rs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            PropertyValueNameToEnumMapper::from_data(data),
+        )))
+    }
+
+    impl ICU4XPropertyValueNameToEnumMapper {
+        /// Get the property value matching the given name, using strict matching
+        ///
+        /// Returns -1 if the name is unknown for this property
+        // #[diplomat::rust_link(icu::properties::names::PropertyValueNameToEnumMapperBorrowed::get_strict, FnInStruct)]
+        // #[diplomat::rust_link(icu::properties::names::PropertyValueNameToEnumMapperBorrowed::get_strict_u16, FnInStruct, hidden)]
```
Comment (sffc) [inline ffi/diplomat/src/properties_names.rs:45]: Question: Why are these rust links commented out? (here and below)

**Review by sffc — COMMENTED** [review]
Looks about right

File: ffi/diplomat/src/properties_names.rs (hunk 1 above)
Comment (Manishearth) [reply ffi/diplomat/src/properties_names.rs:45]: oh, must have happened by mistake

Comment (Manishearth) [reply ffi/diplomat/src/properties_names.rs:45]: https://github.com/unicode-org/icu4x/pull/3202

**Pull request merged**